Wrappers module async (#150)
V4Fire/Core: src/core/async/modules/wrappers/README.md

# core/async/modules/wrappers

> This module provides Async wrappers for data-providers and event-emitters.

Reviewer: ```suggestion
This module provides a bunch of helpers to wrap some objects, like event-emitters or data-providers.
```

> ### Data-providers

Reviewer: ```suggestion
## Data-providers
```

> 

Reviewer: The wrapper takes a link to the "raw" data provider and returns a new object that based on the original, but all async methods and properties are wrapped by Async.

```js
import Async from 'core/async';
import Provider from 'core/data';

const
  rawDataProvider = new Provider(),
  watcher = new Async();

> const dp = watcher.wrapDataProvider(rawDataProvider);

Reviewer: Нужно больше примеров, напрмир про объединение групп

dp.method('POST').get('foo');
```

> ### Event-emitters

Reviewer: ```suggestion
## Event-emitters
```

> ```js

Reviewer: Перевод строки после заголовка + описание

The wrapper takes a link to the "raw" event emitter and returns a new object that based on the original, but all async methods and properties are wrapped by Async. Notice, the wrapped methods can take additional Async parameters, like group or label. In addition, the wrapper adds new methods, like "on" or "off", to make the emitter API more standard.

import Async from 'core/async';

const
  watcher = new Async();

const wrappedEventEmitter = watcher.wrapEventEmitter(window);

const handler = () => console.log('scroll event');

> // Alias for wrappedEventEmitter.on

Reviewer: ```suggestion
// Alias for `wrappedEventEmitter.on`
```

wrappedEventEmitter.addEventListener('scroll', handler, {
  capture: true,
  label: 'label',
});

> // Alias for wrappedEventEmitter.off

Reviewer: ```suggestion
// Alias for `wrappedEventEmitter.off`
```
